### PR TITLE
xgboost: 1.7.6 -> 2.0.0

### PR DESCRIPTION
**Pull request by nviets**
## Description of changes

Updating xgboost to 2.0.0.

## Things done

Built library with and without GPU support and tested basic functionality.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

### CODE REVIEW

**Comment by nviets** [comment]
Result of `nixpkgs-review pr 255061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.xgboost</li>
    <li>python310Packages.xgboost.dist</li>
    <li>python311Packages.xgboost</li>
    <li>python311Packages.xgboost.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>xgboost</li>
    <li>xgboostWithCuda</li>
  </ul>
</details>

**Comment by nviets** [comment]
###### Failed derivations

- [ ] python3.10-xgboost-2.0.0.drv (derivation hash: 55ngigrh1f98j7sa19jma6lxzi2fwlfr)
  - https://gist.github.com/nviets/287c89512ec84dd0ed45f90f912d9245
- [ ] python3.11-xgboost-2.0.0.drv (derivation hash: 6hd9k05bvpbj48b08rs9fb5bf5j4qzl9)
  - https://gist.github.com/nviets/db9c5d417a9ac3a808315499d77e0b53

**Comment by nviets** [comment]
Result of `nixpkgs-review pr 255061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>haskellPackages.xgboost-haskell</li>
    <li>haskellPackages.xgboost-haskell.doc</li>
    <li>python310Packages.xgboost</li>
    <li>python310Packages.xgboost.dist</li>
    <li>python311Packages.xgboost</li>
    <li>python311Packages.xgboost.dist</li>
    <li>xgboost</li>
    <li>xgboostWithCuda</li>
  </ul>
</details>

**Comment by nviets** [comment]
###### Failed derivations

- [ ] python3.10-shap-0.42.1.drv (derivation hash: z3gh8fdd6lnjz9fscyyj59ivcbmlfjl9)
  - https://gist.github.com/nviets/3cd4a7b13a20fc648547d776a968cc38

**Comment by nviets** [comment]
@evax, @natsukium - I would like to update xgboost to 2.0.0, but python-module/shap breaks. I am not sure if it was broken already or simply not ready for xgboost 2.0.0. Please see the log above where a unit test appears to fail. Let me know if you have any objections, otherwise I am going to open this PR for review and merge. Thanks!

**Comment by nviets** [comment]
There are already some [open issues for shap](https://github.com/shap/shap/issues/3254) regarding xgboost 2.0.0, so I expect it will be resolved soon. I am going ahead with the PR in anticipation that someone can bumpt the version of shap soon.

**Comment by nviets** [comment]
@Mindavi - would you be able to merge? Thank you!

**Comment by Mindavi** [comment]
Can you squash the commits or add xgboost: to the second commit?

File: pkgs/development/python-modules/xgboost/default.nix [hunk 1]
```diff
@@ -4,19 +4,21 @@
...
   pname = "xgboost";
+  format = "pyproject";
   inherit (xgboost) version src meta;
 
   disabled = pythonOlder "3.8";
 
   nativeBuildInputs = [ cmake ];
   buildInputs = [ xgboost ];
-  propagatedBuildInputs = [ numpy scipy ];
+  propagatedBuildInputs = [ numpy scipy hatchling ];
```
Comment (natsukium) [inline pkgs/development/python-modules/xgboost/default.nix:21]: I think hatchling is build-time dependency, so it should go to nativeBuildInputs.

**Comment by nviets** [comment]
@Mindavi - squashed the commits and addressed feedback from @natsukium. Ready to merge hopefully. Thanks!

**Comment by nviets** [comment]
Result of `nixpkgs-review pr 255061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python310Packages.xgboost</li>
    <li>python310Packages.xgboost.dist</li>
    <li>python311Packages.xgboost</li>
    <li>python311Packages.xgboost.dist</li>
    <li>xgboost</li>
    <li>xgboostWithCuda</li>
  </ul>
</details>

**Comment by Mindavi** [comment]
Probably also good to mark broken or not supported on Darwin, whichever fits better. I see ofborg fails to build it

**Comment by nviets** [comment]
Thanks @Mindavi. Upstream indicates this build should work on Darwin, so I used the broken flag for now. I'm opening an issue to track it.

**Review by Mindavi — APPROVED** [review]

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 255061` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>xgboostWithCuda</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python310Packages.xgboost</li>
    <li>python310Packages.xgboost.dist</li>
    <li>python311Packages.xgboost</li>
    <li>python311Packages.xgboost.dist</li>
    <li>xgboost</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Comment by nviets** [comment]
Thanks @natsukium! The darwin failure is tracked here: https://github.com/NixOS/nixpkgs/issues/257752.

**Pull request merged**